packages/commerce-sdk-react/src/hooks/types.ts

```diff
@@ -208,3 +209,14 @@ export type CacheUpdateMatrix<Client extends ApiClient> = {
         ? CacheUpdateGetter<MergedOptions<Client, Argument<Client[Method]>>, Data>
         : never
 }
+
+type CustomEndpointArg = Parameters<typeof helpers.callCustomEndpoint>[0]
+type CustomEndpointArgClientConfig = Parameters<
+    typeof helpers.callCustomEndpoint
+>[0]['clientConfig']
+// The commerce-sdk-isomorphic custom endpoint helper REQUIRES clientConfig as mandatory argument
+// But we inject the configs for users from the provider, so this custom type is created
+// to make clientConfig optional when calling useCustomQuery/useCustomMutation
+export type CustomEndpointArgClientConfigOptional = Omit<CustomEndpointArg, 'clientConfig'> & {
```

Comment:
Should this be more readable `OptionalClientConfigFromCustomEndpoint` or `OptionalCustomEndpointClientConfig`?

Reply:
I think this makes sense. I guess my question is, if the user chooses to pass in the client config, as the type allows, will that be used an take precedence over the value that we inject?

Reply:
To me I find the naming confusing since it is long, it sounds like this is about `CustomEndpointArg` but it is more about `ClientConfig` for customEndpointArg that is optional. I think we should put the word ClientConfig first in the name to make it clear

Reply:
thanks for the suggestion!